crates/inspection-service/src/inspection_service.rs

```diff
@@ -231,3 +242,113 @@ pub fn start_inspection_service(node_config: NodeConfig) {
             .unwrap();
     });
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use futures::executor::block_on;
+    use hyper::body;
+    use once_cell::sync::Lazy;
+    use prometheus::{register_int_counter, IntCounter};
+    use std::io::read_to_string;
+
+    // This metrics counter only exists in this test context; and the rest of the system's metrics counters _don't_ exist, so we need to add one here for test_inspect_metrics() below.
+    const INT_COUNTER_NAME: &str = "INT_COUNTER";
+    pub static INT_COUNTER: Lazy<IntCounter> =
+        Lazy::new(|| register_int_counter!(INT_COUNTER_NAME, "An integer counter").unwrap());
+
+    // exercise the serve_requests() handler kinda like how the HTTP framework would
+    fn do_test_get(config: &NodeConfig, path: &str) -> Response<Body> {
+        let mut uri = String::from("http://127.0.0.1:9201");
+        uri += path;
+        block_on(serve_requests(
+            Request::builder()
+                .uri(uri)
+                .method(Method::GET)
+                .body(Body::from(""))
+                .unwrap(),
+            config.clone(),
+        ))
+        .unwrap()
+    }
+
+    #[test]
+    fn test_inspect_configuration() {
+        let mut config = NodeConfig::get_default_validator_config();
+
+        config.inspection_service.expose_configuration = false;
+        let mut r1 = do_test_get(&config, CONFIGURATION_PATH);
```

Comment:
nit: this might be subjective, but a lot of our code uses underscore naming conventions, and we generally try to be more explicit than less, e.g.:
- `r1` -> `response_1` (or `resp_1`, etc.)
- `r1_body` -> `resp_1_body`
- `r2bs` -> `resp_2_body_str`

This just makes things more uniform 😄

Reply:
yeah, just these variables are only test code and only exist for a moment and they don't really matter, ... </whine>, done